### PR TITLE
Global styles controllers: extend with WP_REST_Posts_Controller and WP_REST_Revisions_Controller

### PR DESCRIPTION
**Pull request by ramonjd**
This change: 

- extends `WP_REST_Global_Styles_Revisions_Controller` with `WP_REST_Revisions_Controller`
- extends `WP_REST_Global_Styles_Controller` with `WP_REST_Posts_Controller`

As part of this work, the PR also:

- Removes shared public methods
- Updates schema to use parent and parent controller's properties. Unsets unnecessary properties.
- Updates relevant tests. Deletes dupes.

## Testing

Head over to the site editor, make a few changes to global styles.

https://github.com/WordPress/wordpress-develop/assets/6458278/39d779fa-7b12-43aa-87be-52726aaec050

1. Ensure that the revisions appear in the global styles revisions panel.
2. Check that you can apply them to the editor canvas

You can test the responses in the editor as well by using the API in the browser console.

```
// Get all post types and check that `wp_global_styles` post type does not support `autosave`
const postTypes = await wp.data.resolveSelect( 'core' ).getPostTypes();
postTypes.find( item => item.slug === 'wp_global_styles' ).supports // {title: true, editor: true, revisions: true}

// Get user global styles
await wp.data.resolveSelect( 'core' ).getEntityRecord( 'root', 'globalStyles', await wp.data.resolveSelect( 'core' ).__experimentalGetCurrentGlobalStylesId() )

// Get global styles items (variations)
await wp.data.resolveSelect( 'core' ).__experimentalGetCurrentThemeGlobalStylesVariations()

// Get all revisions.
await wp.data.resolveSelect( 'core' ).getRevisions( 'root', 'globalStyles', await wp.data.resolveSelect( 'core' ).__experimentalGetCurrentGlobalStylesId() ) 

// Get a single revision, where <REVISION_ID> is the id of the single revision.
await wp.data.resolveSelect( 'core' ).getRevision( 'root', 'globalStyles', await wp.data.resolveSelect( 'core' ).__experimentalGetCurrentGlobalStylesId(), <REVISION_ID> ) 
```

Trac ticket: https://core.trac.wordpress.org/ticket/60131

### CODE REVIEW

**Comment by ramonjd** [comment]
Closing in favour of https://github.com/WordPress/wordpress-develop/pull/6105

File: tests/phpunit/tests/rest-api/rest-global-styles-revisions-controller.php [hunk 1]
```diff
@@ -590,13 +591,13 @@ public function test_get_items_invalid_max_pages_should_error() {
...
 	 *
-	 * Duplicate of WP_Test_REST_Revisions_Controller::test_get_items_default_query_should_fetch_all_revisions
+	 * Duplicate of WP_Test_REST_Revisions_Controller::test_get_items_default_query_should_fetch_all_revisons
 	 *
 	 * @ticket 58524
 	 *
 	 * @covers WP_REST_Global_Styles_Controller::get_items
 	 */
-	public function test_get_items_default_query_should_fetch_all_revisions() {
+	public function test_get_items_default_query_should_fetch_all_revisons() {
```
Comment (swissspidy) [inline tests/phpunit/tests/rest-api/rest-global-styles-revisions-controller.php]: Why add a typo?

```suggestion
	public function test_get_items_default_query_should_fetch_all_revisions() {
```

File: tests/phpunit/tests/rest-api/rest-global-styles-revisions-controller.php [hunk 2]
```diff
@@ -590,13 +591,13 @@ public function test_get_items_invalid_max_pages_should_error() {
 	/**
 	 * Tests that the default query should fetch all revisions.
 	 *
-	 * Duplicate of WP_Test_REST_Revisions_Controller::test_get_items_default_query_should_fetch_all_revisions
+	 * Duplicate of WP_Test_REST_Revisions_Controller::test_get_items_default_query_should_fetch_all_revisons
```
Comment (swissspidy) [inline tests/phpunit/tests/rest-api/rest-global-styles-revisions-controller.php]:
```suggestion
	 * Duplicate of WP_Test_REST_Revisions_Controller::test_get_items_default_query_should_fetch_all_revisions
```

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
Committed in https://core.trac.wordpress.org/changeset/58225

File: tests/phpunit/tests/rest-api/rest-global-styles-revisions-controller.php (hunk 1 above)
Comment (ramonjd) [reply tests/phpunit/tests/rest-api/rest-global-styles-revisions-controller.php]: Good catch. No reason, human error I'd wager

**Pull request closed without merging**